src/github.com/getlantern/flashlight/client/reverseproxy.go

```diff
@@ -10,12 +10,25 @@ import (
 	"github.com/getlantern/flashlight/proxy"
 )
 
+// getReverseProxy waits for a message from client.rpCh to arrive and then it
+// writes it back to client.rpCh before returning it as a value. This way we
+// always have a balancer at client.rpCh and, if we don't have one, it would
+// block until one arrives.
 func (client *Client) getReverseProxy() *httputil.ReverseProxy {
+
+	// This rpChMu will protect rpCh and ensure it always have, at most, one
+	// element enqueued.
+	client.rpChMu.Lock()
+	defer client.rpChMu.Unlock()
```

Comment:
I think this is also unnecessary unless we have two concurrent initializations (and in that case, that should be prevented explicitly).